apps/api/src/common/authority-checker.service.ts

```diff
@@ -68,13 +67,15 @@ export class AuthorityCheckerService {
       prisma
     )
 
-    // Check if the user has the authority to perform the action
-    if (
-      !permittedAuthorities.has(authority) &&
-      !permittedAuthorities.has(Authority.WORKSPACE_ADMIN)
-    ) {
+    const hasRequiredAuthority = authorities.some(
+      (auth) =>
+        permittedAuthorities.has(auth) ||
```

Comment:
As per the new logic, we would need all the authorities passed into `authorities` to be contained in `permittedAuthorites`. So, some operation wont work. 

You can add these two steps to improve the performance:
- Check for workspace admin role before the loop
- Use set operations for authority check (IDK how that can be implemented, but feel free to skip it if its too challenging.)

Reply:
sure I'll try to do it 🫡

Reply:
@anaskhan28 any progress on this?

Reply:
I was busy with college will probably try to complete it by tomorrow 🫡

Reply:
@anaskhan28 hey! would you please be able to make the fixes? I would really love to merge the changes!